### ISSUE TITLE
Document nix::sys::wait Module

### ISSUE THREAD

**Issue by mmstick** [nix-rust/nix#643]
I'm trying to use `waitpid` in Ion, but I cannot figure out what each of the `WaitStatus` variants and their fields represent.

**Comment by Susurrus** [comment]
Yeah, pretty much everything in `src/sys/wait.rs` needs docs. A lot of this should be discernible by reading the man pages on `waitpid`, as we're just providing a safe interface to that functionality. The main functionality should be identical.

**Comment by mmstick** [comment]
Sadly, the man pages for `waitpid` aren't enough for me to understand how to use it. What's the difference between each of the variants? For example, what's the difference between `Continued` and `StillAlive`, or `Stopped` and `Signaled`?

**Comment by Susurrus** [comment]
`Continued` probably maps to the `WIFCONTINUED` `wstatus` value. `StillAlive` probably means it hasn't received any signals. `Stopped` and `Signaled` should correspond to `WIFSTOPPED` and `WIFSIGNALED` respectively. those `WIF*` constants are described in the `waitpid` man pages for more details. I'm not familiar with `waitpid`, so hopefully that gives you a little direction. Let me know if you get stuck again.

**Comment by mmstick** [comment]
I'm going to work on documenting this module, after having implemented it in the Ion shell and figuring it out.
